src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,6 +31,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
```

Comment:
Delegate validation to the constituent classes. For exampke Block should check validity of the Block part of the address string.